bag_unpacker/src/test/scala/uk/ac/wellcome/platform/archive/bagunpacker/Archive.scala

```diff
@@ -0,0 +1,109 @@
+package uk.ac.wellcome.platform.archive.bagunpacker
+
+import java.io._
+
+import org.apache.commons.compress.archivers.{
+  ArchiveOutputStream,
+  ArchiveStreamFactory
+}
+import org.apache.commons.compress.compressors.{
+  CompressorOutputStream,
+  CompressorStreamFactory
+}
+import org.apache.commons.io.IOUtils
+
+import scala.concurrent.Future
+import scala.util.Try
+
+class Archive(
+  archiverName: String,
+  compressorName: String,
+  outputStream: OutputStream
+) {
+
+  private val compress = compressor(compressorName)(_)
+  private val pack = packer(archiverName)(_)
+
+  private val compressorOutputStream =
+    compress(outputStream)
+
+  private val archiveOutputStream = pack(
+    compressorOutputStream
+  )
+
+  def finish() = synchronized {
+    Future.fromTry(Try {
```

Comment:
Why are we wrapping this in `Future.fromTry(Try {`? Am I missing something, or isn’t this equivalent to `Future {`?